submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,173 @@
+const BODY = document.body;
+const MAIN = document.querySelector(".main");
+const FILTERS = document.querySelector(".filters");
+const RESET_BUTTON = document.querySelector(".filters__button");
+const DAY_NIGHT = document.querySelector(".day_night");
+const FRIENDS_COUNT = 60;
+const REQUEST_LINK = `https://randomuser.me/api/?results=${FRIENDS_COUNT}`;
+
+const fetchData = async (requestLink) => {
+	try {
+		const response = await fetch(requestLink);
+		if (!response.ok) {
+			throw Error("HTTP-Error: " + response.statusText);
+		}
+		const { results } = await response.json();
+		return results;
+	} catch (error) {
+		alert(error);
+	}
+};
+
+const initApp = (friendsDataArray) => {
+	const friends = new FriendsList(friendsDataArray, MAIN);
+	friends.renderCards();
+	bindEventListeners(friends);
+};
+
+const bindEventListeners = (friends) => {
+	FILTERS.addEventListener("input", ({ target }) => {
+		if (target.name === "search") {
+			friends.filterBySearch(target.value);
+		}
+		if (target.name === "gender") {
+			friends.filterByGender(target.id);
+			FILTERS.elements.search.value = "";
+			const checkedUserName = FILTERS.querySelector("[name=userName]:checked");
+			if (checkedUserName) {
+				friends.sortByName(checkedUserName.id);
+			}
+			const checkedUserAge = FILTERS.querySelector("[name=userAge]:checked");
+			if (checkedUserAge) {
+				friends.sortByAge(checkedUserAge.id);
+			} //this two forEach callbacks is needed to re-sort new arrays of cards, filtered by gender
+		}
+		if (target.name === "userName") {
+			friends.sortByName(target.id);
+			FILTERS.elements.search.value = "";
+			const checkedUserAge = FILTERS.querySelector("[name=userAge]:checked");
+			if (checkedUserAge) {
+				checkedUserAge.checked = false;
+			}
+		}
+		if (target.name === "userAge") {
+			friends.sortByAge(target.id);
+			FILTERS.elements.search.value = "";
+		}
+	});
+	RESET_BUTTON.addEventListener("click", (e) => {
+		e.preventDefault();
+		FILTERS.querySelectorAll("input[type=radio]:checked").forEach((radioButton) => {
+			radioButton.checked = false;
+		});
+		FILTERS.elements.search.value = "";
+		friends.resetFilters();
+	});
+	FILTERS.addEventListener("submit", (event) => {
+		if (event.key === "Enter") {
```

Comment:
You don't need this check when you preventing submit event, am I right?

Reply:
oops, I was testing the submit event, forgot to change back.